Build failure with liblzf on Windows

### Solution to issue cannot be found in the documentation.

- [x] I checked the documentation.

### Issue

```
2024-10-04T15:54:17.8519798Z CMake Error at D:/bld/open3d_1728056681741/_h_env/Library/lib/cmake/liblzf/liblzf-targets.cmake:91 (message):
2024-10-04T15:54:17.8521351Z   The imported target "liblzf::liblzf" references the file
2024-10-04T15:54:17.8521970Z 
2024-10-04T15:54:17.8522543Z      "D:/bld/open3d_1728056681741/_h_env/Library/lib/lzf.lib"
2024-10-04T15:54:17.8523038Z 
2024-10-04T15:54:17.8523605Z   but this file does not exist.  Possible reasons include:
2024-10-04T15:54:17.8524030Z 
2024-10-04T15:54:17.8528059Z   * The file was deleted, renamed, or moved to another location.
2024-10-04T15:54:17.8528339Z 
2024-10-04T15:54:17.8528656Z   * An install or uninstall procedure did not complete successfully.
2024-10-04T15:54:17.8530147Z 
2024-10-04T15:54:17.8530520Z   * The installation package was faulty and contained
2024-10-04T15:54:17.8530697Z 
2024-10-04T15:54:17.8531270Z      "D:/bld/open3d_1728056681741/_h_env/Library/lib/cmake/liblzf/liblzf-targets.cmake"
2024-10-04T15:54:17.8531895Z 
2024-10-04T15:54:17.8532204Z   but not all the files it references.
2024-10-04T15:54:17.8532565Z 
2024-10-04T15:54:17.8532844Z Call Stack (most recent call first):
2024-10-04T15:54:17.8537207Z   D:/bld/open3d_1728056681741/_h_env/Library/lib/cmake/liblzf/liblzf-config.cmake:17 (include)
2024-10-04T15:54:17.8537843Z   3rdparty/find_dependencies.cmake:293 (find_package)
2024-10-04T15:54:17.8538226Z   3rdparty/find_dependencies.cmake:783 (open3d_find_package_3rdparty_library)
2024-10-04T15:54:17.8538574Z   CMakeLists.txt:507 (include)
```

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1046393&view=logs&j=e5cdccbf-4751-5a24-7406-185c9d30d021&t=0c979b57-e97b-5e0a-363d-f964823a2ab0&l=1489



### Installed packages

See: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1046393&view=logs&j=e5cdccbf-4751-5a24-7406-185c9d30d021&t=0c979b57-e97b-5e0a-363d-f964823a2ab0

### Environment info

See: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1046393&view=logs&j=e5cdccbf-4751-5a24-7406-185c9d30d021&t=0c979b57-e97b-5e0a-363d-f964823a2ab0

--- a/recipe/CMakeLists.txt
+++ b/recipe/CMakeLists.txt
@@ -52,6 +52,11 @@ set_target_properties(${PROJECT_NAME} PROPERTIES
 	SOVERSION ${API_VERSION_MAJOR}
 	OUTPUT_NAME ${LIBRARY_NAME}
 )
+if(MSVC)
+	set_target_properties(${PROJECT_NAME} PROPERTIES
+		WINDOWS_EXPORT_ALL_SYMBOLS TRUE
+	)
+endif()
 add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
 
 add_library(embed INTERFACE)

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -15,7 +15,7 @@ source:
   - path: liblzf.pc.in
 
 build:
-  number: 2
+  number: 3
   run_exports:
     - {{ pin_subpackage('liblzf', max_pin='x.x') }}
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -32,6 +32,7 @@ test:
     - test -f $PREFIX/lib/pkgconfig/{{ name }}.pc  # [unix]
     - test -f ${PREFIX}/lib/cmake/{{ name }}/{{ name }}-config.cmake  # [unix]
     - if not exist %PREFIX%\\Library\\bin\\lzf%SHLIB_EXT% exit 1  # [win]
+    - if not exist %PREFIX%\\Library\\lib\\lzf.lib exit 1  # [win]
     - if not exist %PREFIX%\\Library\\lib\\pkgconfig\\{{ name }}.pc exit 1  # [win]
     - if not exist %PREFIX%\\Library\\lib\\cmake\\{{ name }}\\{{ name }}-config.cmake exit 1  # [win]
 

